### PR TITLE
Add tests for ember-console

### LINKED ISSUE #1: tests

### PR DESCRIPTION
**Pull request by kratiahuja**
Fixes #1 

- [x] Refactored `ember console` so that fastboot API is into its own file. Will make it easier in future to extend `ember console` for other platforms.
- [x] Added an edge error case
- [x] Added basic test harness.

cc: @stefanpenner

### CODE REVIEW

File: lib/utils/fastboot-console.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            return app.visit(...arguments);
+          },
+          get sandbox() {
+            return app._app.sandbox.sandbox;
+          },
+          get instance() {
+            return response.instance;
+          }
+      };
+      resolve({app, response, fastbootConsoleApi})
```
Comment (stefanpenner) [inline lib/utils/fastboot-console.js:47]: we can avoid the outer promise, but simply the pojo here:

```js
return { app, response, fastbootConsoleApi }
````

File: lib/utils/fastboot-console.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    }, (error) => {
+      reject(error)
```
Comment (stefanpenner) [inline lib/utils/fastboot-console.js:49]: and then the reject can be avoided entirely

**Review by stefanpenner — CHANGES_REQUESTED** [review]
Awesome!  Left one nit-pick, otherwise this looks great.

**Comment by kratiahuja** [comment]
Just realized travis god isn't running new tests. Will update config of both gods in a while.

**Comment by kratiahuja** [comment]
Sorry @stefanpenner haven't got time to work on this. I'll work on fixing this this weekend.

File: lib/utils/fastboot-console.js (hunk 1 above)
Comment (kratiahuja) [reply lib/utils/fastboot-console.js:47]: The reason why I want the outer promise and the rejection is, I want to extend ember-console to help user understand their fastboot error. I have some ideas that I want to iterate on in a later PR :)

**Comment by kratiahuja** [comment]
@stefanpenner super sorry for the delay here. totally forgot about this.

Updated to make both gods happy. Also upgrade ember-cli version in the addon and setup appveyor to run tests as well.

**Review by stefanpenner — APPROVED** [review]

**Comment by stefanpenner** [comment]
you rock!

**Pull request merged**